src/hash.c

```diff
@@ -82,6 +83,7 @@ void module_sha_complete(struct image *image, uint8_t *hash)
 void ri_sha256(struct image *image, unsigned int offset, unsigned int size,
 	       uint8_t *hash)
 {
+	assert(size < (int64_t)image->adsp->image_size - offset);
```

Comment:
When possible and when it makes sense, I think it's more readable to avoid subtractions:
```shell
assert((uint64_t) offset + size < image->adsp->image_size)
```
This order is also closer to the argument order of the `module_sha_update()` call immediately below.

Reply:
I focus on negative 'size', it why there was little non-standard order, but you're right, I can just extend temporary value to uin64_t to achieve it. Moreover changed condition to `<=`